File: src/test/testData/dart/ExceptionStatements.dart
class Tests {
  @complexity(1)
  void tryDoesNotAddToComplexityNorNesting() {
    try {
      if (true) {                       // +1
        parseFile("salary.txt");
      }
    } on Exception {
      // empty - no points if no actual handler body branching
    }
  }

  @complexity(4)
  void catchAddsToBoth(dynamic log) {
    try {
      parseFile("salary.txt");
    } on RangeError catch (e) {           // +1 catch
      if (e.toString() == "Not found") {  // +2 (nesting=1)
        log.warn(e.toString());
      }
    } on Exception catch (e) {            // +1 catch
      log.error(e.toString());
    }
  }

  @complexity(1)
  void catchWithGenericException() {
    try {
      rethrowSomething("abc");
    } on Object catch (e) {               // +1 catch
      print(e);
    }
  }
}

void parseFile(String path) {}
void rethrowSomething(String s) {}

File: src/test/testData/kotlin/ExceptionStatements.kt
@Complexity(3)
fun catchAddsComplexityAndNesting(a: Boolean) {
    try {
        println(1)
    } catch (e: Exception) {    // +1
        if (a) {                // +2 (nesting=1)
            println(2)
        }
    }
}

@Complexity(1)
fun finallyIsFree(a: Boolean) {
    try {
        if (a) {                // +1
            println(1)
        }
    } finally {                 // +0
        println(2)
    }
}
